sygnal/apnspushkin.py

```diff
@@ -551,14 +552,20 @@ def _get_payload_full(
         if loc_args:
             payload["aps"].setdefault("alert", {})["loc-args"] = loc_args
 
-        if badge is not None:
+        if self.get_config("push_with_badge", bool, True) and badge is not None:
```

Comment:
Rather than pulling this option value out of the config on each request, please do so in `__init__` and set to a class variable, then reference that class variable instead. i.e.:

```py
def __init__(self, name: str, sygnal: "Sygnal", config: Dict[str, Any]) -> None:
    # ...
    self.push_with_badge = self.get_config("push_with_badge", bool, True)
```

```py
def _get_payload_event_id_only(
    self, n: Notification, default_payload: Dict[str, Any]
) -> Dict[str, Any]:
    # ...
    if self.push_with_badge and badge is not None:
```

Reply:
:+1: will be in the next commit